src/proxy/http/profiles.rs

```diff
@@ -294,50 +311,69 @@ pub mod router {
         }
     }
 
+    type ConcreteRouter<T, Svc, B> =
+        rt::Router<http::Request<B>, ConcreteDstRecognize<T>, rt::FixedMake<T, Svc>>;
+
     #[derive(Debug)]
-    pub struct Layer<G, M, R, B> {
+    pub struct Layer<G, M, R, RBody, MBody> {
         get_routes: G,
         route_layer: svc::Builder<R>,
         suffixes: Vec<dns::Suffix>,
         /// This is saved into a field so that the same `Arc`s are used and
         /// cloned, instead of calling `Route::default()` every time.
         default_route: Route,
-        _p: ::std::marker::PhantomData<fn() -> (M, B)>,
+        _p: ::std::marker::PhantomData<fn() -> (M, RBody, MBody)>,
     }
 
     #[derive(Debug)]
-    pub struct MakeSvc<G, M, R, B> {
+    pub struct MakeSvc<G, M, R, RBody, MBody> {
         inner: M,
         get_routes: G,
         route_layer: svc::Builder<R>,
         suffixes: Vec<dns::Suffix>,
         default_route: Route,
-        _p: ::std::marker::PhantomData<fn(B)>,
+        _p: ::std::marker::PhantomData<fn(RBody, MBody)>,
     }
 
-    pub struct Service<G, T, R, B>
+    pub struct Service<G, T, R, S, RMk, M, RBody, MBody>
     where
-        T: WithRoute + Clone,
+        T: WithAddr + WithRoute + Clone + Eq + Hash,
         T::Output: Clone + Eq + Hash,
-        R: rt::Make<T::Output>,
-        R::Value: svc::Service<http::Request<B>> + Clone,
+        M: rt::Make<T>,
+        M::Value: svc::Service<http::Request<MBody>> + Clone,
+        R: svc::Layer<S, Service = RMk>,
+        RMk: rt::Make<T::Output>,
+        RMk::Value: svc::Service<http::Request<RBody>> + Clone,
     {
         target: T,
-        stack: R,
+        inner: M,
+        route_layer: svc::Builder<R>,
         route_stream: Option<G>,
```

Comment:
We call this field `route_stream` in `struct Service`, and `get_routes` in `MakeSvc`. I see it is an `Option<..>` here, but could we use the same name? I think it helps for readability.

Reply:
I'm inclined to leave this alone as it wasn't touched in this PR.